dashboard/src/reducers/kube.ts

```diff
@@ -0,0 +1,34 @@
+import { LOCATION_CHANGE, LocationChangeAction } from "connected-react-router";
+import * as _ from "lodash";
+
+import { getType } from "typesafe-actions";
+import actions from "../actions";
+import { KubeAction } from "../actions/kube";
+import { IKubeState } from "../shared/types";
+
+const initialState: IKubeState = {
+  items: {},
+};
+
+const kubeReducer = (
+  state: IKubeState = initialState,
+  action: KubeAction | LocationChangeAction,
+): IKubeState => {
+  switch (action.type) {
+    case getType(actions.kube.requestResource):
+      return { ...state, items: _.merge({}, state.items, action.payload) };
+    case getType(actions.kube.receiveResource):
+      return { ...state, items: _.merge({}, state.items, action.payload) };
```

Comment:
It is very strange that the reducer has all the steps generic (via action.payload) instead of defining them here. It also makes them more difficult to test, or need to test it on a higher level (dispatcher level not the store level)

I'd personally set the different changes here and having as only common payload content `the resource`.

So for example, the reducer will be in charge of doing this `{ isFetching: false, error: e }` after getting the resource and the error.

BTW, this is the reason why I was confused about the tests, because all the reducer cases are the same.